src/Language/Haskell/Exts/InternalLexer.hs

```diff
@@ -368,8 +368,9 @@ isPragmaChar c = isAlphaNum c || c == '_'
 -- Used in the lexing of type applications
 -- Why is it like this? I don't know exactly but this is how it is in
 -- GHC's parser.
+-- Symbol `:` is added so that `@:` is lexed as `VarSym "@:"`. See Issue #326
```

Comment:
GHC's counterpart [`notFollowedBySymbol`](https://github.com/ghc/ghc/blob/2f453414c5b1b27b6a84a3b5c7cd1bb5449f1f5d/compiler/parser/Lexer.x#L1000) does not contain `:`.
```haskell
notFollowedBySymbol :: AlexAccPred ExtsBitmap
notFollowedBySymbol _ _ _ (AI _ buf)
  = nextCharIsNot buf (`elem` "!#$%&*+./<=>?@\\^|-~")
```

There are two rules that matches for `@:` in Lexer.x.
```alex
  -- See Note [Lexing type applications]
<0> {
    [^ $idchar \) ] ^
  "@"
    / { ifExtension TypeApplicationsBit `alexAndPred` notFollowedBySymbol }
    { token ITtypeApp }
}
```
and
```
<0> {
  @varsym                                          { varsym }
  ...
}
```
Then, alex chooses the rule that consumes the longes prefix. Thus it is recognized as an operator "@:".
> When the input stream matches more than one rule, the rule which matches the longest prefix of the input stream wins. If there are still several rules which match an equal number of characters, then the rule which appears earliest in the file wins.

(Cited from [3.2. Syntax of Alex files](https://www.haskell.org/alex/doc/html/alex-files.html#rules))

This patch may be ad-hoc. Nonetheless it is correct because `isOpSymbol` is only used for TypeApplications.